### ISSUE TITLE
CNTK uninstall leaves marks in vcxproj which need to be manually removed

### ISSUE THREAD

**Issue by mia-at-speechmorphing** [Microsoft/CNTK#2252]
Hello,
When I installed **CNTK CPU only** I see that there are lines added to my vcxproj:

 <ImportGroup Label="ExtensionTargets">
    <Import Project="..\..\..\..\packages\CNTK.CPUOnly.2.0.0\build\native\CNTK.CPUOnly.targets" Condition="Exists('..\..\..\..\packages\CNTK.CPUOnly.2.0.0\build\native\CNTK.CPUOnly.targets')" />
  </ImportGroup>

and

<Error Condition="!Exists('..\..\..\..\packages\CNTK.CPUOnly.2.0.0\build\native\CNTK.CPUOnly.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\..\..\..\packages\CNTK.CPUOnly.2.0.0\build\native\CNTK.CPUOnly.targets'))" />

When I uninstalled **CNTK CPU Only**, however, the error condition was not removed from my vcxproj file, and similarly, the import for CNTK.CPUOnly remained in the vcxproj file after I uninstalled CNTK.CPUOnly

had to manually remove then - especially with the error element in place link stage fails as it tries to find the (obviously) missing package.
I believe cleanup would be in place here.

**Comment by wolfma61** [comment]
can you give us some details...
what does talk about CNTK CPU only mean ... what are you talking about ? 
the nuget package?
which version?
what error condition was kept in the project file? 

thank you for giving us the detauils

Wolfgang

**Comment by mia-at-speechmorphing** [comment]
Thank you for the quick reply, Wolfgang.
I work in VisualStudio 2015 community version 14.0.25431.01 Update 3, Microsoft .Net framework version 4.7.02046

I downloaded CNTK.CPUOnly version 2.0.0 from NuGet (via Visual Studio) and it worked well.
Now I wanted to get CNTK.GPU
Steps:

  1.  go to NuGet solution and uninstall CNTK.CPUOnly version from appropriate project
  2.  install CNTK.GPU (version 2.1.0)
  3.  build the project.

When I build, I get error message that CNTK.CPUOnly files are missing from the link.

So I looked into my project vcxproj file, and I saw these lines which I reported.
I removed them manually and now build is completed without error and my program works.

Note: when uninstalling CNTK.CPUOnly, all folders and DLLs for that version were successfully removed. The only “leftover” were the lines in vcxproj file.

Please let me know if I can give more information.

Best regards
Mia

**Comment by wolfma61** [comment]
cool - thank you ...
we'll take a look ;)
meaning: will try to repro and figure out what is going on!

**Comment by mia-at-speechmorphing** [comment]
Thank you, and good luck!

**Comment by zhouwangzw** [comment]
I tried to uninstall CPU and re-install GPU Nuget packages using VS2015 Enterprise Update 3, as well as using VS2017 Community version (I do not have VS2015 Community version). Both VS work as expected: the package information in .vcxproj file has been removed after uninstall, and the new package is added after new install. Same for uninstall CPU and reinstall CPU packages. 

Might be the case that the vcxproj was opened by other process when uninstalling the Nuget package, so the file was not updated correctly?
